File: src/test/java/com/baulsupp/oksocial/i9n/AuthorizationTest.java
package com.baulsupp.oksocial.i9n;

import com.baulsupp.oksocial.Main;
import com.google.common.collect.Lists;
import ee.schimke.oksocial.output.TestOutputHandler;
import okhttp3.Response;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class AuthorizationTest {
  private Main main = new Main();
  private TestOutputHandler<Response> output = new TestOutputHandler<Response>();
  private TestCredentialsStore credentialsStore = new TestCredentialsStore();

  {
    main.outputHandler = output;
    main.credentialsStore = credentialsStore;
  }

  @Test public void setToken() throws Exception {
    main.authorize = true;
    main.token = "abc";
    main.arguments = Lists.newArrayList("test");

    main.run();

    assertEquals("abc", credentialsStore.tokens.get("localhost"));
  }

  @Test public void authorize() throws Exception {
    main.authorize = true;
    main.arguments = Lists.newArrayList("test");

    main.run();

    assertEquals("testToken", credentialsStore.tokens.get("localhost"));
  }

  @Test public void authorizeByHost() throws Exception {
    main.authorize = true;
    main.arguments = Lists.newArrayList("https://test.com/test");

    main.run();

    assertEquals("testToken", credentialsStore.tokens.get("localhost"));
  }

  @Test public void authorizeWithArgs() throws Exception {
    main.authorize = true;
    main.arguments = Lists.newArrayList("test", "TOKENARG");

    main.run();

    assertEquals("TOKENARG", credentialsStore.tokens.get("localhost"));
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/i9n/AuthorizationTest.kt
package com.baulsupp.oksocial.i9n

import com.baulsupp.oksocial.Main
import com.baulsupp.oksocial.output.TestOutputHandler
import okhttp3.Response
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class AuthorizationTest {
  private val main = Main()
  private val output = TestOutputHandler<Response>()
  private val credentialsStore = TestCredentialsStore()

  init {
    main.outputHandler = output
    main.credentialsStore = credentialsStore
  }

  @Test
  fun setToken() {
    main.authorize = true
    main.token = "abc"
    main.arguments = mutableListOf("test")

    main.run()

    assertEquals("abc", credentialsStore.tokens["localhost"])
  }

  @Test
  fun authorize() {
    main.authorize = true
    main.arguments = mutableListOf("test")

    main.run()

    assertEquals("testToken", credentialsStore.tokens["localhost"])
  }

  @Test
  fun authorizeByHost() {
    main.authorize = true
    main.arguments = mutableListOf("https://test.com/test")

    main.run()

    assertEquals("testToken", credentialsStore.tokens["localhost"])
  }

  @Test
  fun authorizeWithArgs() {
    main.authorize = true
    main.arguments = mutableListOf("test", "TOKENARG")

    main.run()

    assertEquals("TOKENARG", credentialsStore.tokens["localhost"])
  }
}
